Source/SwiftLintBuiltInRules/Rules/Style/AttributeNameSpacingRule.swift

```diff
@@ -0,0 +1,144 @@
+            Example("@available↓ (*, deprecated)"): Example("@available(*, deprecated)"),
+            Example("@MyPropertyWrapper↓ (param: 2) "): Example("@MyPropertyWrapper(param: 2) "),
+            Example("nonisolated↓ (unsafe) var _value: X?"): Example("nonisolated(unsafe) var _value: X?"),
+            Example("@MyProperty↓ ()"): Example("@MyProperty()"),
+            Example("""
+            let closure1 = { @MainActor↓ (a, b) in
+            }
+            """): Example("""
+            let closure1 = { @MainActor(a, b) in
+            }
+            """),
+        ]
+    )
+}
+
+private extension AttributeNameSpacingRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: DeclModifierSyntax) {
+            guard node.detail != nil, node.name.trailingTrivia.isNotEmpty else {
+                return
+            }
+
+            addViolation(
+                startPosition: node.name.endPositionBeforeTrailingTrivia,
+                endPosition: node.name.endPosition,
+                replacement: ""
+            )
+        }
+
+        override func visitPost(_ node: AttributeSyntax) {
+            // Check for trailing trivia after the '@' sign
+            // Handles cases like `@ MainActor` / `@ escaping`
+            if node.atSign.trailingTrivia.isNotEmpty {
+                addViolation(
+                    startPosition: node.atSign.endPositionBeforeTrailingTrivia,
+                    endPosition: node.atSign.endPosition,
+                    replacement: ""
+                )
+            }
+
+            let hasTrailingTrivia = node.attributeName.trailingTrivia.isNotEmpty
+
+            // Handles cases like @MyPropertyWrapper (param: 2)
+            if node.arguments != nil, hasTrailingTrivia {
+                addViolation(
+                    startPosition: node.attributeName.endPositionBeforeTrailingTrivia,
+                    endPosition: node.attributeName.endPosition,
+                    replacement: ""
+                )
+            }
+
+            if !hasTrailingTrivia, node.isEscaping {
+                // Handles cases where escaping has the wrong spacing: `@escaping()`
+                addViolation(
+                    startPosition: node.attributeName.endPositionBeforeTrailingTrivia,
+                    endPosition: node.attributeName.endPosition,
+                    replacement: " "
+                )
+            }
+        }
+
+        private func addViolation(
+            startPosition: AbsolutePosition,
+            endPosition: AbsolutePosition,
+            replacement: String
+        ) {
+            let correction = ReasonedRuleViolation.ViolationCorrection(
+                start: startPosition,
+                end: endPosition,
+                replacement: replacement
+            )
+
+            violations.append(at: endPosition, correction: correction)
```

Comment:
The description of this rule is very detailed which is good. It's impractical as a violation message, though. Better to set an explicit reason here that depends on the actual violation and is much shorter. That would look like:
```swift
violations.append(
    .init(
        position: endPosition, 
        reason: reason, 
        severity: configuration.severity, 
        correction: .init(
            start: startPosition,
            end: endPosition,
            replacement: replacement
        )
    )
)
```

Incidentally, this is also why the tests fail.

Reply:
Updated!